pkg/tsdb/cloudwatch/models/api.go

```diff
@@ -21,8 +22,10 @@ type RequestContext struct {
 	MetricsClientProvider MetricsClientProvider
 	LogsAPIProvider       CloudWatchLogsAPIProvider
 	OAMAPIProvider        OAMAPIProvider
+	EC2APIProvider        EC2APIProvider
 	Settings              CloudWatchSettings
 	Features              featuremgmt.FeatureToggles
+	RegionCache           *sync.Map
```

Comment:
So it is technically used in the old path here: https://github.com/grafana/grafana/blob/main/pkg/tsdb/cloudwatch/metric_find_query.go#L45

I'm using a feature flag to preserve the old path just in case I'm misunderstanding something and we want to keep the old regions handler while we revert/fix the change. 

When I look at the code it seems to only cache the region on instance.Settings.Profile as a cachekey but my understanding is that instance.Settings.Profile is often blank, IIRC it's only set when using the credentials file provider? 

It kind of feels to me like if we only sometimes use a regions cache, it's probably not necessary to preserve. If we someday decide we're making too many api calls to fetch region it feels like the kind of thing we could add back a cache for and ensure it's used regardless of authentication method. 

But again it's totally possible I'm misreading this or something so I wanted the option to keep the old behavior if it turns out there is something amiss, so I added back the RegionCache but only use it if the feature flag is set to off.

Reply:
[This regionCache](https://github.com/grafana/grafana/blob/dd8f88b194e0aedd98e5b51ced7a088e03681f6f/pkg/tsdb/cloudwatch/cloudwatch.go#L116) is what I'm seeing being used still.

The one "RegionCache" seems to have been introduced here and seems to be unused --maybe an artifact from some refactoring? Let me know if I have missed something!

Reply:
yikes my bad thank you for spotting!